### ISSUE TITLE
CEQA project geometries

### ISSUE THREAD

**Issue by max-mapper** [RadicalResearchLLC/CEQA_tracker#1]
Hi, I am looking into a volunteer project for the California Native Plant Society (my local chapter is in Pasadena) to scrape ceqanet and notify local CNPS chapters to alert them of potential projects. Showing projects on a map wasn't something I had considered, but I just found this repo on a github search and the warehouse shiny app is pretty awesome. I had a couple questions if you have time. 

1. How did you query only warehouse projects from ceqanet? 
2. How did you extract parcel numbers and then get project boundaries from ceqanet?

Thank you!

**Comment by RadicalResearchLLC** [comment]
Hi Max-mapper,

1.  It was manually checking each project, nothing automated.  If naming was consistent enough in project descriptions, the CEQANET data could be keyword searched for a couple dozen terms, but it would still need manual QA/QC.  
2. There's a hodgepodge of stuff, but I've manually entered hundreds of lat-longs, and recently developed a dashboard to help draw the polygons and export them by project number.  I can share the code on github in a separate repository if you are interested.  

I've worked recently with the IE chapter of the CNPS and Aaron Echols and I'd be happy to collaborate or share what I've done and see what kind of auto-notification stuff is possible.

**Comment by max-mapper** [comment]
Cool! Thanks for the reply. I was impressed you had the Irwindale Brew Yard project on the map... we are trying to protect the habitat adjacent to that proposed project (there are rare Horkelia growing there among other things). Looking at your map, it looks like one of the (if not the) largest proposed warehouse projects in LA County! I did not know that.

CNPS has this new project to try and help members understand CEQA https://ct.cnps.org/. I'm new to CEQA as well and trying to understand which document types and queries from ceqanet matter for early detection of potentially damaging project proposals. If you have any thoughts on that I'd appreciate it, I only just started digging around today. The initial idea was to download all the NOP (Notice Of Preparation) projects, maybe filter them by project size if that is machine readable, and then have monthly emails to chapters in each county.

**Comment by RadicalResearchLLC** [comment]
That's a cool website.  I like the filters for the different phases of the project.

I'd note that NOP is first stage only for projects the lead agency has deemed controversial.  Oftentimes a lead agency will consider a project as an MND with no NOP stage. It is important for folks to get eyes on those projects that might be misclassified to force them to do a proper EIR when they omit or misrepresent the situation on the ground.

**Comment by max-mapper** [comment]
Ahh makes sense, thank you.

**Comment by max-mapper** [comment]
I ended up using pdftotext and setting up a google cloud run pipeline that checks all uploaded PDFs every night and scans for mentions of rare native plant species on MND or NOP documents https://github.com/max-mapper/ceqa-plants/

**Comment by RadicalResearchLLC** [comment]
Interesting - but it assumes that the consultants for the project will do their job.

I just dealt with a project where they missed multiple rare species, and I imagine most won't mention rare species in the NOP/MND summary text.

**Comment by max-mapper** [comment]
Yes I'll probably expand the word list... this is just a prototype. I have a list of all 6000+ native plants, or I could just search for all of the genera to broaden it to any report that mentions any plant by scientific name. And probably a manual word list with other high value keyword like "survey period" or "endemic" etc. If you want to run it to scan PDFs for a custom keyword list it deploys pretty easily to google cloud run.

**Comment by RadicalResearchLLC** [comment]
Max,

How is your CEQA plant tracker working out?  Center for Biological Diversity is trying to do something similar and I was wondering if your solution would help them.
https://github.com/max-mapper/ceqa-plants/

Mike McCarthy, PhD
Radical Research LLC

**Comment by max-mapper** [comment]
Hi Mike,

Life took a bit of a turn. We lost our home in the Eaton Fire and my wife
and I decided it would be best to raise our kids in Tokyo, where we are now
living. Unfortunately I am not working actively on any U.S. based projects
anymore. I had been talking with Nick Jensen at CNPS about the CEQA tracker
project and he had discussed it with Jun Bando, but ultimately they didn't
have the funding on hand at the time and then the fire happened and I
couldn't pursue it further. They may be interested in helping out on it, so
I would encourage the folks at CBD to check in with Nick. My code was at
the level of proof of concept and is still running every day and I am happy
to answer questions.
